config.go

```diff
@@ -263,17 +263,17 @@ func (d *DaemonConfig) ServerTLS() *tls.Config {
 	return nil
 }
 
-// SetupDaemonConfig returns a DaemonConfig object as configured by reading the provided config file
-// and environment.
-func SetupDaemonConfig(logger *logrus.Logger, configFile string) (DaemonConfig, error) {
+// SetupDaemonConfig returns a DaemonConfig object that is the result of merging the lines
+// in the provided configFile and the environment variables. See `example.conf` for all available config options and their descriptions.
+func SetupDaemonConfig(logger *logrus.Logger, configFile io.Reader) (DaemonConfig, error) {
```

Comment:
This function really really shouldn't accept a concrete instance of a `logrus.Logger`. This should be an interface and, if consumers want, they can pass their own implementation (or a NoOpLogger)/

Reply:
That sounds like a great idea! Thank you!

Reply:
Agreed on the Logrus concrete type thing.  Would be a nice addition to Gubernator v3 to ditch Logrus once and for all.